docs/2.guide/5.best-practices/performance.md

```diff
@@ -0,0 +1,305 @@
+---
+navigation.title: 'Nuxt Performance'
+title: Nuxt performance
+description: Best practices for improving performance of Nuxt apps.
+---
+
+Nuxt comes with built-in features designed to improve your application's performance and contribute to better [Core Web Vitals](https://web.dev/articles/vitals). There are also multiple Nuxt core modules that assist in improving performance in specific areas. This guide outlines best practices to optimize performance of your Nuxt application.
+
+## Built-in Features
+
+Nuxt offers several built-in features that help you optimize performance of your website. Understanding how these features work is crucial for achieving blazingly-fast performance.
+
+### Links
+
+[`<NuxtLink>`](/docs/api/components/nuxt-link) is a drop-in replacement for both Vue Router's `<RouterLink>` component and HTML's `<a>` tag. It intelligently determines whether the link is internal or external and renders it accordingly with available optimizations (prefetching, default attributes, etc.)
+
+```html
+<template>
+  <NuxtLink to="/about">About page</NuxtLink>
+</template>
+
+<!-- Which will render to with Vue Router & Smart Prefetching -->
+<a href="/about">About page</a>
+```
+
+Nuxt automatically includes smart prefetching. That means it detects when a link is visible (by default), either in the viewport or when scrolling and prefetches the JavaScript for those pages so that they are ready when the user clicks the link.
+
+You can also opt for prefetching on interaction instead:
+
+```ts
+export default defineNuxtConfig({
+  experimental: {
+    defaults: {
+      nuxtLink: {
+        prefetchOn: 'interaction',
+      },
+    }
+  }
+})
+```
+
+:read-more{title="NuxtLink" to="/docs/api/components/nuxt-link"}
+
+### Hybrid Rendering
+
+In more complex applications, we may need a full control over how our application is rendered to support cases where some pages could be generated at build time, while others should be client-side rendered
+
+Hybrid rendering allows different caching rules per route using Route Rules and decides how the server should respond to a new request on a given URL:
+
+```ts
+export default defineNuxtConfig({
+  routeRules: {
+    '/': {
+      prerender: true
+    },
+    '/products/**': {
+      swr: 3600
+    },
+    '/blog': {
+      isr: 3600
+    },
+    '/admin/**': {
+      ssr: false
+    },
+  }
+})
+```
+
+Nuxt server will automatically register corresponding middleware and wrap routes with cache handlers using Nitro caching layer.
+
+:read-more{title="Hybrid rendering" to="/docs/guide/concepts/rendering#hybrid-rendering"}
+
+### Lazy Loading Components
+
+To dynamically import a component (also known as lazy-loading a component) all you need to do is add the Lazy prefix to the component's name. This is useful if the component is not always needed.
```

Comment:
I found it confusing when using this in the docs. It seems adding the 'Lazy' prefix to the componet's name is not enough, the component still needs to use v-if with the ref, otherwise it won't be lazy loaded. 
Just doing this won't lazy load the component:
` <LazyMountainsList />`

I wrote about this here as personal notes:
https://nuxt-loading-guide.vercel.app/components